rcl/src/rcl/logging_rosout.c

```diff
@@ -250,13 +251,19 @@ rcl_ret_t rcl_logging_rosout_init_publisher_for_node(rcl_node_t * node)
   if (rcutils_hash_map_key_exists(&__logger_map, &logger_name)) {
     // @TODO(nburek) Update behavior to either enforce unique names or work with non-unique
     // names based on the outcome here: https://github.com/ros2/design/issues/187
+    const char * node_name = rcl_node_get_name(node);
+    if (NULL == node_name) {
+      RCL_SET_ERROR_MSG("Node name was null.");
+      return RCL_RET_ERROR;
+    }
```

Comment:
I think we should do this error handling a bit differently.  If we fail to get it, we can still go to the `RCUTILS_LOG_WARN_NAMED` line below, we will just have an "unknown" name.  So maybe something like:

```suggestion
    if (NULL == node_name) {
      node_name = "unknown node";
    }
```

(to be perfectly honest if this ever fails, there will probably be other failures shortly after, but we may as well handle this as best we can)